spyder/utils/environ.py

```diff
@@ -9,6 +9,8 @@
 """
 
 # Standard library imports
+from functools import cache
```

Comment:
```suggestion
from functools import lru_cache
```

Sorry, the `cache` decorator was added in Python 3.9, so we need to use `lru_cache` instead.